src/test/java/com/google/step/youtube/YouTubeClientBuilder.java

```diff
@@ -0,0 +1,148 @@
+
+    public static YouTube create() throws IOException {
+        YouTube mockYouTubeClient = mock(YouTube.class);
+
+        Channels mockChannels = mock(YouTube.Channels.class);
+        mockListChannels = mock(YouTube.Channels.List.class);
+        when(mockYouTubeClient.channels()).thenReturn(mockChannels);
+        when(mockChannels.list("contentDetails")).thenReturn(mockListChannels);
+        when(mockListChannels.setId(anyString())).thenReturn(mockListChannels);
+        when(mockListChannels.setForUsername(anyString())).thenReturn(mockListChannels);
+
+        PlaylistItems mockPlaylistItems = mock(YouTube.PlaylistItems.class);
+        mockListPlaylistItems = mock(YouTube.PlaylistItems.List.class);
+        when(mockYouTubeClient.playlistItems()).thenReturn(mockPlaylistItems);
+        when(mockPlaylistItems.list("snippet")).thenReturn(mockListPlaylistItems);
+        when(mockListPlaylistItems.setMaxResults(YouTubeInfoScraper.MAX_PLAYLIST_RESULTS))
+                .thenReturn(mockListPlaylistItems);
+        when(mockListPlaylistItems.setPlaylistId(anyString())).thenReturn(mockListPlaylistItems);
+
+        Videos mockVideos = mock(YouTube.Videos.class);
+        mockListVideos = mock(Videos.List.class);
+        when(mockYouTubeClient.videos()).thenReturn(mockVideos);
+        when(mockVideos.list("snippet")).thenReturn(mockListVideos);
+        when(mockListVideos.setId(anyString())).thenReturn(mockListVideos);
+        when(mockListVideos.setFields("items(id, snippet(publishedAt, title, description))"))
+                .thenReturn(mockListVideos);
+
+        Search mockSearch = mock(Search.class);
+        mockListSearch = mock(Search.List.class);
+        when(mockYouTubeClient.search()).thenReturn(mockSearch);
+        when(mockSearch.list("snippet")).thenReturn(mockListSearch);
+        when(mockListSearch.setMaxResults(YouTubeInfoScraper.MAX_SEARCH_RESULTS)).thenReturn(mockListSearch);
+        when(mockListSearch.setQ(anyString())).thenReturn(mockListSearch);
+
+        return mockYouTubeClient;
+    }
+
+    public static PromoCode newPromoCode(String description) {
+        return PromoCode.create(PROMOCODE, description, VIDEO_ID, VIDEO_TITLE, DATE);
+    }
+
+    public Channels.List getMockListChannels() {
+        return mockListChannels;
+    }
+
+    public PlaylistItems.List getMockListPlaylistItems() {
+        return mockListPlaylistItems;
+    }
+
+    public Videos.List getMockListVideos() {
+        return mockListVideos;
+    }
+
+    public Search.List getMockListSearch() {
+        return mockListSearch;
+    }
+
+    public ChannelListResponse newBasicChannelResponse() {
+        ChannelListResponse testChannelResponse = new ChannelListResponse();
+        Channel channel = new Channel();
+        channel.setContentDetails(
+                new ChannelContentDetails().setRelatedPlaylists(new RelatedPlaylists().setUploads(UPLOAD_ID)));
+        testChannelResponse.setItems(Arrays.asList(channel));
```

Comment:
Could move the declaration on line 109 to be inlined in line 112 so a new variable doesn't have to be declared.